File: Medium/2026-05-25-1871-Jump-Game-VII/solution.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        n = len(s)
        if s[n - 1] != '0':
            return False
        
        reachable = 0
        for i in range(n):
            if i > reachable:
                return False
            if s[i] == '0':
                if i + minJump <= n - 1:
                    reachable = max(reachable, i + maxJump)
        
        return reachable >= n - 1

File: Medium/2026-05-25-1871-Jump-Game-VII/solution.java
class Solution {
    public boolean canReach(String s, int minJump, int maxJump) {
        int n = s.length();
        if (s.charAt(n - 1) != '0') return false;

        int maxReach = 0;
        for (int i = 0, lastReachable = 0; i < n; i++) {
            if (i > maxReach) return false;
            if (s.charAt(i) == '0') {
                if (i >= minJump) {
                    lastReachable = Math.max(lastReachable, i + minJump);
                }
                if (i <= maxReach) {
                    maxReach = Math.max(maxReach, i + maxJump);
                }
            }
        }
        return maxReach >= n - 1;
    }
}